[profiles] Added `[system_tools]` (#2989)
conan-io/docs: reference/config_files/profiles.rst

Given this situation, it could make sense to want to use your already installed CMake version, so it's enough to declare
it as a ``system_tools`` in your profile (``default`` one or any other in use):

.. code-block:: text
    :caption: *myprofile*

    ...

    [system_tools]
    cmake/3.24.2

Whenever you want to create the package, you'll see that build requirement is already satisfied because of the system tool
declaration:

.. code-block:: bash
    :emphasize-lines: 9,18

    $ conan create . -pr myprofile --build=missing
    ...
    -------- Computing dependency graph --------
    Graph root
        virtual
    Requirements
        pkg/2.0#3488ec5c2829b44387152a6c4b013767 - Cache
    Build requirements
        cmake/3.24.2 - System tool

    -------- Computing necessary packages --------

    -------- Computing necessary packages --------
    pkg/2.0: Forced build from source
    Requirements
        pkg/2.0#3488ec5c2829b44387152a6c4b013767:20496b332552131b67fb99bf425f95f64d0d0818 - Build
    Build requirements
        cmake/3.24.2 - System tool

Notice that if the ``system_tools`` declared does not make a strict match with the ``tool_requires`` one (version or

> version range), then Conan will try to bring them remotely or locally as usual. Given the previous example, changing the

Reviewer: Maybe we can give an example with ranges?

    Read more about tool requires in this section: :ref:`consuming_packages_tool_requires`.

.. _reference_config_files_profiles_system_tools:

[system_tools]
+++++++++++++++

.. include:: ../../common/experimental_warning.rst

This section is similar to the previous one, **[tool_requires]**, but it's intended to list only the tool requires

> that are already in your own system and you don't want Conan to search them remotely or locally.

Reviewer 2: ```suggestion
that are already in your own system and you don't want Conan to search for, neither remotely nor locally.
```
